client/app/components/DateInput.jsx

```diff
@@ -10,24 +11,35 @@ import 'antd/lib/date-picker/style/index.less';
 function DateInput({
   value,
   onSelect,
+  // eslint-disable-next-line react/prop-types
   clientConfig,
+  // eslint-disable-next-line react/prop-types
+  $scope,
 }) {
   const format = clientConfig.dateFormat || 'YYYY-MM-DD';
-  const defaultValue = moment(value, format);
+  const additionalAttributes = {};
+  if (value && value.isValid()) {
+    additionalAttributes.defaultValue = value;
+  }
+  if (isFunction(onSelect)) {
+    additionalAttributes.onChange = (v) => {
+      onSelect(v);
+      $scope.$applyAsync();
```

Comment:
I think it's better to keep Angular code outside of the React components as much as possible, to make it easier to transition.

Also, in this case -- all the different Date/Time inputs eventually call the same function (`param.setValue`), so it's even easier to introduce this change in a single place. Btw, I wouldn't do this in the `Parameter#setValue` method but rather have a `setValue` function in the `Parameters` component which takes care of this.